crates/cxx-qt-gen/src/syntax/foreignmod.rs

```diff
@@ -112,50 +113,43 @@ fn verbatim_to_foreign_type(tokens: &TokenStream) -> Result<Option<ForeignItemTy
     .parse2(tokens.clone())
 }
 
-pub struct ForeignFnSelf {
-    pub ident: Ident,
-    pub typ: syn::Type,
-}
+pub fn self_type_from_foreign_fn(signature: &Signature) -> Result<Receiver> {
+    if let Some(FnArg::Receiver(receiver)) = signature.inputs.iter().next() {
+        if !receiver.attrs.is_empty() {
+            return Err(Error::new(
+                receiver.span(),
+                "Attributes on the `self:` receiver are not supported",
+            ));
+        }
 
-pub fn self_type_from_foreign_fn(signature: &Signature) -> Result<ForeignFnSelf> {
-    let mut span: &dyn ToTokens = signature;
-    if let Some(arg) = signature.inputs.iter().next() {
-        span = arg;
-        if let syn::FnArg::Typed(PatType {
-            pat,
-            ty,
-            attrs,
-            colon_token: _,
-        }) = arg
-        {
-            if !attrs.is_empty() {
-                return Err(Error::new_spanned(
-                    arg,
-                    "Attributes on the `self:` receiver are not supported",
-                ));
-            }
-            if let syn::Pat::Ident(PatIdent {
-                ident,
-                // It should be a `self:` value, without `mut` or `&`
-                mutability: None,
-                by_ref: None,
-                attrs,
-                subpat: _,
-            }) = &**pat
-            {
-                if ident == "self" && attrs.is_empty() {
-                    return Ok(ForeignFnSelf {
-                        ident: ident.clone(),
-                        typ: *ty.clone(),
-                    });
-                }
-            }
+        if receiver.mutability.is_some() {
+            return Err(Error::new(
+                receiver.span(),
+                "mut on the `self:` receiver are not supported",
```

Comment:
```suggestion
                "mut on the `self:` receiver is not supported",
```

Reply:
Shouldn't this be "mut on self (i.e. `&mut self`) are not supported, use `self: Pin<&mut T>` instead" ?

Reply:
Ah, yes, good point :+1:

Reply:
Done